agent/xds/clusters.go

```diff
@@ -849,6 +849,31 @@ func (s *ResourceGenerator) configIngressUpstreamCluster(c *envoy_cluster_v3.Clu
 	if threshold != nil {
 		c.CircuitBreakers.Thresholds = []*envoy_cluster_v3.CircuitBreakers_Thresholds{threshold}
 	}
+
+	// Configure the default outlier detector for upstream service
+	outlierDetection := ToOutlierDetection(cfgSnap.IngressGateway.Defaults.PassiveHealthCheck)
+
+	// override the default outlier detection value
+	if svc.PassiveHealthCheck != nil {
+		if svc.PassiveHealthCheck.Interval != 0 {
+			outlierDetection.Interval = durationpb.New(svc.PassiveHealthCheck.Interval)
+		}
+		if svc.PassiveHealthCheck.MaxFailures != 0 {
+			outlierDetection.Consecutive_5Xx = &wrappers.UInt32Value{Value: svc.PassiveHealthCheck.MaxFailures}
+		}
+
+		// if svc.PassiveHealthCheck.EnforcingConsecutive5xx != nil && *svc.PassiveHealthCheck.EnforcingConsecutive5xx != 0 {
+		if svc.PassiveHealthCheck.EnforcingConsecutive5xx != nil {
+			outlierDetection.EnforcingConsecutive_5Xx = &wrappers.UInt32Value{Value: *svc.PassiveHealthCheck.EnforcingConsecutive5xx}
+		}
+	}
+
+	// Specail handling for failover peering service, which has set MaxEjectionPercent
+	if c.OutlierDetection != nil && c.OutlierDetection.MaxEjectionPercent != nil {
+		outlierDetection.MaxEjectionPercent = &wrappers.UInt32Value{Value: c.OutlierDetection.MaxEjectionPercent.Value}
+	}
+
+	c.OutlierDetection = outlierDetection
```

Comment:
I think always setting outlier detection will end up being a breaking change and is causing golden tests like [this one](agent/xds/testdata/clusters/ingress-with-chain-and-failover-to-cluster-peer.latest.golden) to change. Is that intended?

If so, we should make this a breaking change in the changelog and we can't backport it.

Reply:
After second thought, I think adding empty outlier_detection make it consistent with other places, like connect-proxy, mesh-gateway

https://github.com/hashicorp/consul/blob/4a32070210cdbff8c457334ee2e3e798b9f263b1/agent/xds/testdata/clusters/connect-proxy-with-tls-outgoing-cipher-suites.latest.golden#L21-L24

https://github.com/hashicorp/consul/blob/4a32070210cdbff8c457334ee2e3e798b9f263b1/agent/xds/testdata/clusters/mesh-gateway-with-exported-peered-services-http-with-router.latest.golden#L17-L19

So I added a note of this breaking change to the changelog.

Reply:
Since this is going to be a breaking change, could you remove the backport labels from this PR?

Reply:
Sure, backport labels are removed.